File: ios/Classes/PlayerMethodManager.swift
//
//  PlayerMethodManager.swift
//  Runner
//
//  Created by nguon pisey on 3/8/21.
//

import Foundation

class PlayerMethodManager {
    
    func methodChannel(binaryMessenger:FlutterBinaryMessenger,handler:@escaping FlutterMethodCallHandler){
        let methodChannelPlayer = FlutterMethodChannel(name: Constant.METHOD_CHANNEL_PLAYER,binaryMessenger:binaryMessenger)
        methodChannelPlayer.setMethodCallHandler(handler)
    }
    
    func eventChannelPlayer(binaryMessenger:FlutterBinaryMessenger,handler:(FlutterStreamHandler & NSObjectProtocol)?){
        let eventChannelPlayer = FlutterEventChannel(name: Constant.EVENT_CHANNEL_PLAYER, binaryMessenger: binaryMessenger)
        eventChannelPlayer.setStreamHandler(handler)
    }

}

File: android/src/main/kotlin/com/pisey/flutter_native_player/PlayerMethodManager.kt
package com.pisey.flutter_native_player

import com.pisey.flutter_native_player.constants.Constant
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel


class PlayerMethodManager {

    fun methodChannel(binaryMessage:BinaryMessenger,handler: MethodChannel.MethodCallHandler){
        MethodChannel(binaryMessage, Constant.METHOD_CHANNEL_PLAYER).setMethodCallHandler(handler)
    }

    fun eventChannel(binaryMessage: BinaryMessenger,handler: EventChannel.StreamHandler){
        EventChannel(binaryMessage, Constant.EVENT_CHANNEL_PLAYER).setStreamHandler(handler)
    }
}